### PR TITLE
allow the passing of custom options for get calls if those options ar…

### PR DESCRIPTION
**Pull request by ibmuserpublic**
…e provided

Currently, the GraphqlConnector does not allow users to pass in custom options during get calls. This small change allows them to do so without impacting users who are calling get without providing those options.

### CODE REVIEW

**Comment by ryanomackey** [comment]
@ibmuserpublic Thanks for the contribution! Just curious though, what's the use case here? Do you have an example?

**Comment by ibmuserpublic** [comment]
@ryanomackey We are having the graphql service make an API call to another service that is outside of the blue zone. This means that we must include certain certifications in the options portion of the API call. It just so happens that, with get calls in graphql, one cannot pass their own key value pairs to options so we cannot add the certs. This small change to gramps should allow users to pass custom keys to options or not without breaking other team's get calls. Please let me know if any further clarification is needed.

**Comment by ibmuserpublic** [comment]
@ryanomackey Hello. I see that there are a few other outstanding PR's on this project. Is there anything that I missed with this PR or some testing that is blocking a merge to master? I would be glad to contribute anything further that would allow this change to go live.

**Comment by ryanomackey** [comment]
> @ryanomackey We are having the graphql service make an API call to another service that is outside of the blue zone. This means that we must include certain certifications in the options portion of the API call. It just so happens that, with get calls in graphql, one cannot pass their own key value pairs to options so we cannot add the certs. This small change to gramps should allow users to pass custom keys to options or not without breaking other team's get calls. Please let me know if any further clarification is needed.

@ibmuserpublic Hmm, so the options that get set here are just ones that `request-promise` accepts. See [their README](https://www.npmjs.com/package/request-promise#api-in-detail) for more detail. Are you perhaps just trying to set some custom request headers? Because that's already enabled with [this line](https://github.com/gramps-graphql/gramps/blob/master/packages/rest-helpers/src/GraphQLConnector.js#L68).

> @ryanomackey Hello. I see that there are a few other outstanding PR's on this project. Is there anything that I missed with this PR or some testing that is blocking a merge to master? I would be glad to contribute anything further that would allow this change to go live.

Yeah, this project could always use some TLC. The "dependabot" PRs are opened automatically by GitHub to help patch vulnerabilities (pretty much like `npm audit` or `snyk` if you're familiar), but those wouldn't block any other PRs from going in typically. That being said, it does look like the Travis check won't run on this PR, so that's something else to look into.

So yeah, not sure if there's an update we need to make here regarding this PR in particular, but I could probably use some help maintaining the overall project regardless.

**Comment by ibmuserpublic** [comment]
> So yeah, not sure if there's an update we need to make here regarding this PR in particular

@ryanomackey Let me try to explain this issue in more detail.

Here is an example of the options that must be passed to another API to get data back.

 var ngHeaders = {
            'Content-Type': 'application/json',
            'isUserAuthenticated': true,
            'x-ibm-client-secret':NGAPI_CLIENT_SECRET,
            'x-ibm-client-id': NGAPI_CLIENT_ID,
            'Cache-Control': 'no-cache'
        }
        var options = {
            method: "GET",
            rejectUnauthorized: false,
            url: ng_url,
            headers: ngHeaders,
            key: fs.readFileSync(params.IEB_CERT_NAME + '.key'),
            cert: fs.readFileSync(params.IEB_CERT_NAME + '.crt')
        }

That app requires these options exactly as they are. More keys can be added to options but at least these keys are required. Putting these keys in headers will not work here, and that has been tested. The change that I proposed does allow for contact with that other app however. I have gone into gramps in my node modules and made the change in this PR and tested that out. Most users of the graphql microservice do not need to do this however, and they can simply omit the custom options keys when calling GET, as they have been doing.

To summarize, this change should be necessary to allow for adding custom options in GET calls. 

Please let me know if you have further questions.

**Comment by hmedney** [comment]
Hi @ryanomackey, great to see you in here! Just wanted to clarify the ask in case there was any confusion remaining. The PR basically exposes the rest of the underlying [request](https://github.com/request/request) library options to gramps consumers. In our case, we need to set the [TLS cert options](https://github.com/request/request#tlsssl-protocol) to connect to a service that required the connection use that cert.

**Comment by ibmuserpublic** [comment]
@ryanomackey @jlengstorf @kbrandwijk Hey everyone, can I become a contributor on this project in the case that no one is able to review and merge this requested change? This change should be fairly low risk and will not impact other users but the change has high impact for my team.
